src/main/java/com/redhat/cloud/notifications/db/BehaviorGroupResources.java

```diff
@@ -0,0 +1,131 @@
+package com.redhat.cloud.notifications.db;
+
+import com.redhat.cloud.notifications.models.BehaviorGroup;
+import com.redhat.cloud.notifications.models.BehaviorGroupAction;
+import com.redhat.cloud.notifications.models.Bundle;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EventType;
+import com.redhat.cloud.notifications.models.EventTypeBehavior;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import org.hibernate.reactive.mutiny.Mutiny;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+public class BehaviorGroupResources {
+
+    private static final Logger LOGGER = Logger.getLogger(BehaviorGroupResources.class.getName());
+
+    @Inject
+    Mutiny.Session session;
+
+    public Uni<BehaviorGroup> createBehaviorGroup(BehaviorGroup behaviorGroup) {
+        return Uni.createFrom().item(behaviorGroup)
+                .onItem().transform(this::addBundleReference)
+                .onItem().transformToUni(session::persist)
+                .call(session::flush)
+                .replaceWith(behaviorGroup);
+    }
+
+    public Multi<BehaviorGroup> getBehaviorGroups(UUID bundleId, String accountId) {
+        String query = "FROM BehaviorGroup WHERE accountId = :accountId AND bundle.id = :bundleId";
+        return session.createQuery(query, BehaviorGroup.class)
+                .setParameter("accountId", accountId)
+                .setParameter("bundleId", bundleId)
+                .getResultList()
+                .onItem().transformToMulti(Multi.createFrom()::iterable);
+    }
+
+    public Uni<Boolean> updateBehaviorGroup(BehaviorGroup behaviorGroup) {
+        String query = "UPDATE BehaviorGroup SET name = :name, displayName = :displayName WHERE accountId = :accountId AND id = :id";
+        return session.createQuery(query)
+                .setParameter("name", behaviorGroup.getName())
+                .setParameter("displayName", behaviorGroup.getDisplayName())
+                .setParameter("accountId", behaviorGroup.getAccountId())
+                .setParameter("id", behaviorGroup.getId())
+                .executeUpdate()
+                .call(session::flush)
+                .onItem().transform(rowCount -> rowCount > 0);
+    }
+
+    public Uni<Boolean> deleteBehaviorGroup(UUID id) {
+        String query = "DELETE FROM BehaviorGroup WHERE id = :id";
+        return session.createQuery(query)
+                .setParameter("id", id)
+                .executeUpdate()
+                .call(session::flush)
+                .onItem().transform(rowCount -> rowCount > 0);
+    }
+
+    public Uni<Void> createBehaviorGroupAction(UUID behaviorGroupId, UUID endpointId) {
+        // TODO Check behavior group accountId is the good one
```

Comment:
This `TODO` is to prevent updating the global behavior groups?

Reply:
That was a TODO for myself which has already been fixed on my machine. I wanted to make sure there wouldn't be cross-accounts access to the behavior groups.